Sources/SWBTaskConstruction/ProductPlanning/ProductPlanner.swift

```diff
@@ -472,10 +472,6 @@ extension StandardTarget
             }
         }
 
-        createNewPhaseNode("AppIntentsMetadataTaskProducer")
-        taskProducers.append(AppIntentsMetadataTaskProducer(taskProducerContext, sourcesBuildPhase: self.sourcesBuildPhase, resourcesBuildPhase: self.resourcesBuildPhase, phaseStartNodes: [setupEndPhaseNode, buildPhasesEndNode], phaseEndNode: endPhaseNode))
```

Comment:
I think most of the task construction failures you're seeing are because once this segment is replaced by a call into an extension as an unordered post-setup task producer, it loses the buildPhasesEndNode input which is added here. I think we could add a new method like `postBuildPhaseTaskProducers` to replicate the behavior before the task producer was moved